### PR TITLE
R4R: Use new tendermint/libs/db.DB.ReverseIterator API

### PR DESCRIPTION
**Pull request by jaekwon**
See https://github.com/tendermint/tendermint/pull/2913

### CODE REVIEW

File: version.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
 package iavl
 
 // Version of iavl.
-const Version = "0.11.1"
```
Comment (ValarDragon) [inline version.go]: this is a breaking change, so it should be `0.12.0` right?

Comment (ebuchman) [reply version.go]: Is it really? What's breaking in the IAVL here? The GoAPI of the underlying db lib went through breaking change, but doesn't seem like there's a breaking change in this PR?

Comment (ValarDragon) [reply version.go]: Does IAVL only work with a given db implementation? My understanding was that it took in an interface, so others can in principle have supplied their own db implementation here.

Comment (ebuchman) [reply version.go]: Great point

**Pull request merged**